packages/rs-drive-abci/src/abci/handler/mod.rs

```diff
@@ -260,10 +280,22 @@ where
                 TxAction::Unmodified
             } as i32;
 
-            tx_results.push(tx_result);
+            if action != TxAction::Removed as i32 {
+                tx_results.push(tx_result);
+            }
             tx_records.push(TxRecord { action, tx });
         }
 
+        // Add up exceeding transactions to the response
+        tx_records.extend(
+            transactions_exceeding_max_block_size
+                .into_iter()
+                .map(|tx| TxRecord {
+                    action: TxAction::Removed as i32,
```

Comment:
Did we make sure this is the right action that won't remove the transaction from the mempool?

Reply:
Fixed